src/com/serotonin/mango/web/dwr/EventHandlersDwr.java

```diff
@@ -330,6 +332,9 @@ private DwrResponseI18n save(int eventSourceId, int eventTypeRef1,
 		vo.setDisabled(disabled);
 
 		DwrResponseI18n response = new DwrResponseI18n();
+
+		XidUtils.validateXid(response, eventService::isXidUnique, xid, handlerId);
```

Comment:
1. This method should be used in the validate method on EventHandlerVO class; (next line "vo.validate(response)" )
2. The non-xid value should be validated, because then the mechanism that automatically generates xid will not work if the user left the field empty.